src/main/java/com/google/starfish/servlets/UserRegistrationServlet.java

```diff
@@ -0,0 +1,170 @@
+package com.google.starfish.servlets;  
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import java.io.IOException;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.Types;  
+import java.sql.ResultSet;
+import java.sql.Date;
+import javax.servlet.ServletException;  
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Servlet that handles user registration and authentication. */  
+@WebServlet("/user-signin")  
```

Comment:
Should this be renamed to match the file name?

Reply:
Hm I don't think it's strictly necessary but I can make a change so that it's renamed to user-registration.